python/packages/ag-ui/agent_framework_ag_ui/_events.py

```diff
@@ -386,6 +386,17 @@ def _handle_function_result_content(self, content: FunctionResultContent) -> lis
         result_message_id = generate_event_id()
         if isinstance(content.result, dict):
             result_content = json.dumps(content.result)  # type: ignore[arg-type]
+        elif isinstance(content.result, list):
```

Comment:
The exact same logic is duplicated in both files. Should this be extracted into a shared utility function (perhaps in `_utils.py` or `_message_adapters.py`) to make sure we have consistency and reduce the maintenance burden? If the serialization logic needs to change, it would need to be updated in both places.